pkg/cloud/scope/cluster.go

```diff
@@ -193,6 +193,10 @@ func (s *ClusterScope) ControlPlaneLoadBalancerName() *string {
 	return nil
 }
 
+func (s *ClusterScope) ClusterControlPlaneEndpoint() clusterv1.APIEndpoint {
+	return s.Cluster.Spec.ControlPlaneEndpoint
```

Comment:
Is there a specific reason of using Cluster's ControlPlaneEndpoint here instead of AWSCluster's ControlPlaneEndpoint?

Cluster's field gets populated at a later time by Cluster controller using AWSCluster field.

Reply:
I thought we can add the Cluster's field as it will be the last operation to happen before setting cluster to provisioned state, but as discussed offline, will keep it as AWSCluster's endpoint as problems may arise before setting this endpoint in owner cluster and we may end up recreating ELB again.